evals/util.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import base64
+import mimetypes
+import os
+from typing import List
+
+import pandas as pd
+from llama_stack_client.lib.agents.agent import Agent
+from tqdm import tqdm
+
+
+async def get_response_row(agent: Agent, input_query: str) -> str:
+    # single turn, each prompt is a new session
+    session_id = agent.create_session(f"session-{input_query}")
+    response = agent.create_turn(
+        messages=[
+            {
+                "role": "user",
+                "content": input_query,
+            }
+        ],
+        session_id=session_id,
+    )
+
+    async for chunk in response:
+        event = chunk.event
+        event_type = event.payload.event_type
+        if event_type == "turn_complete":
+            return event.payload.turn.output_message.content
+
+
+def data_url_from_file(file_path: str) -> str:
```

Comment:
so I have realized that this isn't a scalable path because URLs have a size limit so we can't arbitrarily push content into a URL. if this is merely a "sample", then we should immediately error out if the max URL length (I believe it is 65536, but please check) is exceeded. and probably also rename the method to sound flaky by default.

Reply:
Yeah, agree that we need better file handling, updated this method name to sound flaky. 

However, I think the URL too long issue (https://github.com/meta-llama/llama-stack/issues/416) don't apply with memory document uploads, since it reads the raw str.